dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java

```diff
@@ -0,0 +1,273 @@
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        when(request.getParameter("host_id")).thenReturn(site.getIdentifier());
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(site);
+        when(session.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)).thenReturn(defaultLang.getId()+"");
+        when(session.getAttribute(WebKeys.PAGE_MODE_SESSION)).thenReturn(PageMode.PREVIEW_MODE);
+
+        // Collection to store attributes keys/values
+        final Map<String, Object> attributes = new ConcurrentHashMap<>();
+
+        // Mock setAttribute
+        Mockito.doAnswer((Answer<Void>) invocation -> {
+            final String key = invocation.getArgumentAt(0, String.class);
+            final Object value = invocation.getArgumentAt(1, Object.class);
+            attributes.put(key, value);
+            return null;
+        }).when(request).setAttribute(Mockito.anyString(), Mockito.anyObject());
+
+        // Mock getAttribute
+        Mockito.doAnswer((Answer<Object>) invocation -> {
+            final String key = invocation.getArgumentAt(0, String.class);
+            return attributes.get(key);
+        }).when(request).getAttribute(Mockito.anyString());
+
+        final String diff = HTMLDiffUtil.htmlDiffPage(pageLive, systemUser, request, response);
+        Assert.assertNotEquals(NOTHING_CHANGED,diff);
+        final Document document = Jsoup.parse(diff);
+        final Elements removedElements = document.select("span.diff-html-removed");
+        for (final Element removedElement : removedElements) {
+           Assert.assertTrue(seekers.contains(removedElement.text()));
+        }
+
+        final Elements addedElements = document.select("span.diff-html-added");
+        for (final Element addedElement : addedElements) {
+            Assert.assertTrue(coneheads.contains(addedElement.text()));
+        }
+    }
+
+
+    /**
+     *Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy od the page but this one isn't modified. So whn compared
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)